 [ActivityManager] Bring up Service if not schedule to restart

Symptom:

A Service which has scheduled to restart, the
ServiceRecord.restartDelay will be set a value which is
greater than 0.

After the Service is restarted, the Service process got
killed again while client application invoked
unbindService(), there may have chances that results
DeadObjectException when schedule UnbindService to the
Service process  in removeConnectionLocked().

In that case, the Service would failed to scheduled restart.
(serviceProcessGoneLocked() -> serviceDoneExecutingLocked()
with finishing = true)

And the Service will unable to bring up afterward
whenever client applications invoked bindService()
because r.restartDelay > 0.

Solution:
Bring up Service if it was also not in the restarting list

diff --git a/services/core/java/com/android/server/am/ActiveServices.java b/services/core/java/com/android/server/am/ActiveServices.java
index a16fcd2..2741051 100644
--- a/services/core/java/com/android/server/am/ActiveServices.java
+++ b/services/core/java/com/android/server/am/ActiveServices.java
@@ -1723,7 +1723,7 @@ private String bringUpServiceLocked(ServiceRecord r, int intentFlags, boolean ex
             return null;
         }
 
-        if (!whileRestarting && r.restartDelay > 0) {
+        if (!whileRestarting && r.restartDelay > 0 && mRestartingServices.contains(r)) {
             // If waiting for a restart, then do nothing.
             return null;
         }
